sympy/vector/coordsysrect.py

```diff
@@ -98,6 +213,65 @@ def __new__(cls, name, location=None, rotation_matrix=None,
             location = Vector.zero
             origin = Point(name + '.origin')
 
+        if transformation is None:
+            if parent is None:
```

Comment:
Why doesn't it make sense? Anyway, the code in the master supports location even without parent, so we need to keep the compatibility.